go/ai/prompt.go

```diff
@@ -426,3 +430,174 @@ func renderDotprompt(templateText string, variables map[string]any, defaultInput
 	}
 	return str, nil
 }
+
+// LoadPromptFolder loads prompts and partials from the input directory for the given namespace
+func LoadPromptFolder(r *registry.Registry, dir string, namespace string) error {
+	useDefaultDir := false
+	if dir == "" {
+		dir = "./prompts"
+		useDefaultDir = true
+	}
+	promptsPath, err := filepath.Abs(dir)
+	if err != nil {
+		errMsg := fmt.Sprintf("Failed to resolve directory: %v", err)
+		if !useDefaultDir {
+			return errors.New(errMsg)
+		}
+		slog.Error(errMsg)
+		return nil
+	}
+	if _, err := os.Stat(promptsPath); os.IsNotExist(err) {
+		errMsg := fmt.Sprintf("Prompts directory does not exist: %s", promptsPath)
+		if !useDefaultDir {
+			return errors.New(errMsg)
+		}
+		slog.Error(errMsg)
+		return nil
+	}
+	return LoadPromptFolderRecursively(r, promptsPath, namespace, "")
+}
+
+// LoadPromptFolderRecursively recursively loads prompts and partials from the directory.
+func LoadPromptFolderRecursively(r *registry.Registry, dir string, namespace string, subDir string) error {
+	parentPath := filepath.Join(dir, subDir)
+	entries, err := os.ReadDir(parentPath)
+	if err != nil {
+		slog.Error("Failed to read directory", "error", err.Error())
+		return nil
```

Comment:
```suggestion
		return fmt.Errorf("failed to read prompt directory structure: %w, err)
```

This is unlikely to happen so it's ok for it to error out. It's probably some permissions issue or something more obscure.